### DISCUSSION TITLE
tileLayer PNG adjust maximum pixel intensity (exposure)

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#5684] [Q&A]
Hi, I'm using the tileLayer with a set of deepzoom pyramid PNG images (16 bit depth) and I would like to know if there is a way I can modify the exposure of the images before they are shown (e.g. set maximum pixel intensity). They are currently underexposed but it would be nice to have a slider that would allow the user to adjust exposure. Would this need to be written in shader code?

**Comment by kylebarron** [comment]
These are the `BitmapLayer` render parameters: https://deck.gl/docs/api-reference/layers/bitmap-layer#render-options

For something more complex you'd probably need to dig into shader code (maybe through a layer extension), or maybe look at luma.gl's shader modules, such as https://luma.gl/docs/api-reference/shadertools/effects-shader-modules

**Reply by cornhundred** [reply]
Thanks @kylebarron I'll check it out

**Reply by cornhundred** [reply]
This [PostProcessEffect](https://github.com/visgl/deck.gl/blob/1049956d723fe7a92c39ea203bc2b5c5fe00cc36/docs/api-reference/core/post-process-effect.md#postprocesseffect) [codepen](https://codepen.io/vis-gl/embed/YbRGvv/?height=450&theme-id=light&default-tab=result) looks like it might do what I want. I'll keep the thread updated if I figure it out.
